src/route.c

```
@@ -0,0 +1,185 @@
+	unsigned int i, msgfilter_mask;
+#endif
+#endif
+
+	if (lctx == NULL)
+		return -1;
+
+	lctx->link_ctx = ctx;
+
+	lctx->link_fd = xsocket(PF_ROUTE, SOCK_RAW | SOCK_CXNB, AF_UNSPEC);
+	if (lctx->link_fd == -1)
+		return -1;
```

Cppcheck (2.20.0)

[error] Memory leak

(memleak)